### ISSUE TITLE
Rendering extra polygons?

### ISSUE THREAD

**Issue by jdrage** [processing/processing#4012]
Hi, I've got a simple patch which seems to render some extra "stuff".

I'm attaching the .pde and a screen shot.

I'm running Processing 3.0 release on Windows 10 on a Lenovo ThinkPad W520 (using the NVIDIA adapter if that matters).

Thanks,
James

**Comment by jdrage** [comment]
![bug_extra_lines](https://cloud.githubusercontent.com/assets/7401782/10560304/444c0444-74bc-11e5-8c11-efb41ccb88e3.PNG)

Hmm. Can't seem to attach a .pde, so I'll paste the source code below.

In the screen shot you can see the dark diagonal lines below the waves, they seem to converge at a common point.

---

/**
 * Bug: Extra lines rendered. 
 */


float t = 0.0;
float[][][] p = new float[80][80][3]; // points for our shape


void setup()
{
  size(1280, 960, P3D);  
  stroke(255, 255);
  noFill();
}

void update()
{
  // Update the time
  t += 0.01;

  float w = 600.0;
  float h = 10.0;
  float in;
  float jn;

  // update the points
  for (int i = 0; i < p[0].length; i++)
  {
    for (int j = 0; j < p.length; j++)
    {
      in = (float)i * w / p.length - w/2;
      jn = (float)j * w / p[0].length - w/2;
      
      p[i][j][0] = jn * 2.0;
      p[i][j][1] = sin(jn + t * 8.0) * h - h/2.0;
      p[i][j][2] = in;
    }
  }
}

void draw()
{  
  background(0);
  scale(2.0);

  update();
  drawCurves();

  // Change the view
  camera(0.0, -height - 10.0 + 5.0 * sin(t * 2.0), (height/2.0) / tan(PI*30.0 / 180.0), 0.0, 0.0, 0.0, 0.0, 1.0, 0.0);
}

void drawCurves()
{
  for (int i = 0; i < p.length; i++)
  {
    beginShape();
    for (int j = 0; j < p[0].length; j++)
    {
      curveVertex(p[i][j][0], p[i][j][1], p[i][j][2]);
    }
    endShape();
  }
}

**Comment by james-mcelwain** [comment]
Hey, I thought your sketch looked fun so I loaded up on my computer (MBP) and didn't see any irregularities (i.e. dark diagonal lines). I'm guessing it has something to do with your hardware--maybe someone smarter than me has some idea.

**Comment by jdrage** [comment]
Thanks for trying it out.

**Comment by JakubValtar** [comment]
This is probably a duplicate of #4007.

**Comment by erniejunior** [comment]
No weird lines here on my Thinkpad as well (using an integrated intel card).

**Comment by JakubValtar** [comment]
I'm getting glitches even on Intel, but I have to rotate it.
```processing
void draw()
{  
  background(0);
  scale(2.0);
  rotateX(map(mouseY, 0, height, -PI, PI));
  rotateY(map(mouseX, 0, width, -PI, PI));
  ...
```
![glitches](https://cloud.githubusercontent.com/assets/3177098/10564220/a86ee03c-75ab-11e5-89f3-ffb448d2d12f.png)

**Comment by jdrage** [comment]
Yes, rotation definitely affects it. Sometimes the endpoints shift to a different "row" in the wave.

**Comment by jdrage** [comment]
Thanks Jakub, that does look like the same issue.

**Comment by erniejunior** [comment]
OK with the rotation I can confirm it too,

**Comment by codeanticode** [comment]
Fixed with https://github.com/processing/processing/commit/ed76e73cb310c1b9287cb42bd167449890f4daee

**Comment by jdrage** [comment]
Thanks for the quick turnaround. I am doing a show on the 29th, I'm hoping to get a release with this fix.

**Comment by jdrage** [comment]
Fixed in 3.0.1. Yay!
